stage0/run.go

```diff
@@ -308,6 +308,15 @@ func Run(cfg RunConfig, dir string) {
 		log.Fatalf("failed changing to dir: %v", err)
 	}
 
+	// we write /etc/machine-id here because systemd-nspawn needs it to link
```

Comment:
It doesn't, if `/etc/machine-id` doesn't exist in the tree and you pass `--link-journal=host` it fails.